user/api.go

```diff
@@ -96,20 +97,21 @@ func InitApi(cfg ApiConfig, store Storage, metrics highwater.Client) *Api {
 	}
 
 	return &Api{
-		Store:            store,
-		ApiConfig:        cfg,
-		metrics:          metrics,
-		logger:           logger,
-		mailchimpManager: mailchimpManager,
+		Store:              store,
+		ApiConfig:          cfg,
+		metrics:            metrics,
+		logger:             logger,
+		mailchimpManager:   mailchimpManager,
+		accessTokenChecker: &AccessTokenChecker{Auth0Domain: cfg.Auth0Domain},
 	}
 }
 
-func (a *Api) AttachPerms(perms clients.Gatekeeper) {
-	a.perms = perms
+func (a *Api) AttachAccessTokenChecker(accessTokenChecker AccessTokenCheckerInterface) {
```

Comment:
Is this just for testing? If so, then I'd recommend the entire `AccessTokenChecker` interface be injected in via the `InitApi` function. The `platform` uses almost all injected interfaces. There is really only one high-level "master" struct per service that creates all of the required interface instances and glues them all together. Works great for testing.